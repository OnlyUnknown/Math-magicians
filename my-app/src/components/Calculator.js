import React from "react";

function Calculator() {
    return (
        <div>
        <Screen />
 
        <Numbers />
        <Arthitics />
        </div>
    )
}

function Screen() {
    return <div>0</div>
}

const Numbers = () => {
    return(
        <div>
                   <Toparth />
    <button>1</button>
    <button>2</button>
    <button>3</button>
    <button>4</button>
    <button>5</button>
    <button>6</button>
    <button>7</button>
    <button>8</button>
    <button>9</button>
    <div>
        <button>0</button>
        <button>.</button>
    </div>
    </div>
    )
}

const Arthitics = () => {
    return(
        <div>
            <button>/</button>
            <button>X</button>
            <button>-</button>
            <button>+</button>
            <button>=</button>
        </div>
    )
}

const Toparth = () => {
    return (<div>
        <button>AC</button>
        <button>+/-</button>
        <button>%</button>
    </div>
    )
}

export default Calculator

