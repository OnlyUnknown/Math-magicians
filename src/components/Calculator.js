import { useState } from 'react';
import PropTypes from 'prop-types';
import calculate from '../logic/calculate';

function Calculator() {
  return (
    <div className="box">
      <Numbers />
    </div>
  );
}



// ButtonName.propTypes = { input: PropTypes.string.isRequired };
// ButtonName.propTypes = { styl: PropTypes.string.isRequired };

const Numbers = () => {
  const [calcData, setCalcData] = useState({
    total: null,
    next: null,
    operation: null,
  });
  const displayValue = calcData.next || calcData.total || '0';
  const handleClick = (buttonName) => {
    const newData = calculate(calcData, buttonName);
    setCalcData(newData);
  };
  const ButtonName = (props) => {
    
  
    const { styl, input } = props;
    return (
      <button onClick={() => handleClick(input)} className={styl} type="button">
        {input}
      </button>
    );
    
  };
  function Screen() {
    return <div className="screen">{displayValue}</div>;
  }
  

  return (
  <>  
  <Screen/>
  <div className="flex-N">
    <div className="Numbers">
      <ButtonName input="AC" />
      <ButtonName input="+/-" />
      <ButtonName input="%" />
      <ButtonName input="1" />
      <ButtonName input="2" />
      <ButtonName input="3" />
      <ButtonName input="4" />
      <ButtonName input="5" />
      <ButtonName input="6" />
      <ButtonName input="7" />
      <ButtonName input="8" />
      <ButtonName input="9" />
      <ButtonName input="0" styl="Zewo" />
      <ButtonName input="." />
    </div>
    <div>
    <div className="arghitics">
        <ButtonName input="รท" />
        <ButtonName input="x" />
        <ButtonName input="-" />
        <ButtonName input="+" />
        <ButtonName input="=" />
      </div>
    </div>

  </div>
  </>
)
};



export default Calculator;
