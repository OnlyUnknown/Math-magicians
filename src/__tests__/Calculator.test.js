import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';
import { render, screen} from '@testing-library/react';
import '@testing-library/jest-dom'

describe('Calculate.js File Testing', () => {
    test("Render by Testing library", () => {
        render(<Calculator/>); 
        const CalcContent = screen.getByTestId("CalcContent");
        expect(CalcContent).toBeInTheDocument()
        })

test('renders correctly', () => {
  const tree = renderer.create(<Calculator />).toJSON();
  expect(tree).toMatchSnapshot();
});
});