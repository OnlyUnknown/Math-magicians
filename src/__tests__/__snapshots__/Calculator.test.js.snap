// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculate.js File Testing renders correctly 1`] = `
<div
  data-testid="CalcContent"
>
  <h3
    className="MathM"
  >
    Let‘s do some math
  </h3>
  <div
    className="box"
  >
    <div
      className="screen"
    >
      0
    </div>
    <div
      className="flex-N"
    >
      <div
        className="Numbers"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          className="Zewo"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          .
        </button>
      </div>
      <div>
        <div
          className="arghitics"
        >
          <button
            onClick={[Function]}
            type="button"
          >
            ÷
          </button>
          <button
            onClick={[Function]}
            type="button"
          >
            x
          </button>
          <button
            onClick={[Function]}
            type="button"
          >
            -
          </button>
          <button
            onClick={[Function]}
            type="button"
          >
            +
          </button>
          <button
            onClick={[Function]}
            type="button"
          >
            =
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
